server/src/main/java/com/broadcom/lsp/domain/cobol/databus/impl/CopybookRepositoryLRU.java

```diff
@@ -40,111 +39,55 @@
 @Singleton
 public class CopybookRepositoryLRU implements CopybookRepository {
   @Getter private final int cacheMaxSize;
-  @Setter @NonNull private ArrayList<CopybookStorable> cpyRepo;
-  private final Comparator<CopybookStorable> storableComparator =
-      Comparator.comparingInt(CopybookStorable::getHit); // Time last recently used
-  private final Comparator<CopybookStorable> timeComparator =
-      Comparator.comparingLong(CopybookStorable::getGenDt);
-
-  @NonNull private final AtomicBoolean isSort = new AtomicBoolean(false);
+  private Cache<String, CopybookModel> cache;
 
   @Inject
   public CopybookRepositoryLRU(@Named("CACHE-MAX-SIZE") int cacheSize) {
     cacheMaxSize = cacheSize;
-    cpyRepo = new ArrayList<>(cacheMaxSize);
-  }
-
-  @Override
-  public synchronized void sortCache() {
-    if (!isSort.get()) {
-      cpyRepo.sort(storableComparator.reversed().thenComparing(timeComparator));
-    }
-    isSort.set(true);
-  }
-
-  @Override
-  public Optional<CopybookStorable> getCopybookStorableFromCache(@NonNull long uuid) {
-    ArrayList<CopybookStorable> shallowCpy = (ArrayList<CopybookStorable>) cpyRepo.clone();
-    Optional<CopybookStorable> cpy =
-        shallowCpy.stream().filter(copy -> uuid == copy.getId()).findAny();
-    if (cpy.isPresent()) return cpy.map(SerializationUtils::clone);
-    return cpy;
+    cache = CacheBuilder.newBuilder()
+        .maximumSize(cacheMaxSize)
+        .expireAfterAccess(3, TimeUnit.HOURS)
+        .build();
   }
 
-  private Optional<CopybookStorable> getCopybookStorableInstance(@NonNull long uuid) {
-    return cpyRepo.stream().filter(copy -> uuid == copy.getId()).findAny();
+  /** For test purpose only */
+  CopybookRepositoryLRU(@Named("CACHE-MAX-SIZE") int cacheSize, Ticker ticker) {
+    cacheMaxSize = cacheSize;
+    cache = CacheBuilder.newBuilder()
+        .maximumSize(cacheMaxSize)
+        .expireAfterAccess(3, TimeUnit.HOURS)
+        .ticker(ticker)
+        .build();
```

Comment:
I think you may extract this builder and reuse the code for both constructors

Reply:
I'm not sure that now it is better.